docs/library/spi_engine/axi_spi_engine.rst

```diff
@@ -0,0 +1,202 @@
+
+Files
+--------------------------------------------------------------------------------
+
+.. list-table::
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Name
+     - Description
+   * - :git-hdl:`master:library/spi_engine/axi_spi_engine/axi_spi_engine.v`
+     - Verilog source for the peripheral.
+   * - :git-hdl:`master:library/spi_engine/axi_spi_engine/axi_spi_engine_ip.tcl`
+     - TCL script to generate the Vivado IP-integrator project for the peripheral.
+
+Configuration Parameters
+--------------------------------------------------------------------------------
+
+.. hdl-parameters::
+
+   * - ASYNC_SPI_CLK
+     - If set to 1 the ``s_axi_aclk`` and ``spi_clk`` clocks are assumed
+       to be asynchronous.
+   * - CMD_FIFO_ADDRESS_WIDTH
+     - Configures the size of the command FIFO.
+   * - SDO_FIFO_ADDRESS_WIDTH
+     - Configures the size of the serial-data out FIFO.
+   * - SDI_FIFO_ADDRESS_WIDTH
+     - Configures the size of the serial-data in FIFO.
+   * - NUM_OFFLOAD
+     - The number of offload control interfaces.
+
+Signal and Interface Pins
+--------------------------------------------------------------------------------
+
+.. hdl-interfaces::
+
+   * - s_axi_aclk
+     - All ``s_axi`` signals and ``irq`` are synchronous to this clock.
+   * - s_axi_aresetn
+     - Synchronous active-low reset.
+       Resets the internal state of the peripheral.
+   * - s_axi
+     - AXI-Lite bus slave.
+       Memory-mapped AXI-lite bus that provides access to modules register map.
+   * - irq
+     - Level-High Interrupt.
+       Interrupt output of the module. Is asserted when at least one of the
+       modules interrupt is pending and unmasked.
+   * - spi_clk
+     - All ``spi_engine_ctrl`` signals and ``spi_resetn`` are
+       synchronous to this clock.
+   * - spi_engine_ctrl
+     - :ref:`spi_engine control-interface` slave.
+       SPI Engine Control stream that contains commands and data for the
+       execution module.
+   * - spi_resetn
+     - This signal is asserted when the module is disabled through the ENABLE
+       register. Typically used as the reset for the SPI Engine modules
+       connected to these modules.
+
+Register Map
+--------------------------------------------------------------------------------
+
+.. hdl-regmap::
+   :name: axi_spi_engine
+
+Theory of Operation
+--------------------------------------------------------------------------------
+
+Typically a software application running on a CPU will be able to execute much
+faster than the SPI engine command will be processed.
+In order to allow the software to execute other tasks while the SPI engine is
+busy processing commands the AXI SPI Engine peripheral offers interrupt-driven
+notification which can be used to notify the software when a SPI command has
+been executed.
+In order to reduce the necessary context switches the AXI SPI Engine peripheral
+incorporates FIFOs to buffer the command as well as the data streams.
+
+FIFOs
```

Comment:
Would it be better to have `FIFOs` and `Synchronization events` as subsections to `Theory of operation`?